### ISSUE TITLE
Custom inspect() method for NodeJS?

### ISSUE THREAD

**Issue by glenjamin** [moment/moment#3229]
Apologies if this is a dupe, I searched but couldn't find anything.

Would you be open to providing an [`inspect()`](https://nodejs.org/dist/latest-v4.x/docs/api/util.html#util_custom_inspect_function_on_objects) method on moment objects which returns a human-readable representation of the moment? This would be useful for debugging and testing purposes.

As well as being used in Node core, a bunch of other inspect libraries out there will detect this method and use it.

I had a bit of a play around in the repl, this seems to be a decent first cut:
```js
// Current
moment()
{ [Number: 1465642464921]
  _isAMomentObject: true,
  _isUTC: false,
  _pf:
   { empty: false,
     unusedTokens: [],
     unusedInput: [],
     overflow: -2,
     charsLeftOver: 0,
     nullInput: false,
     invalidMonth: null,
     invalidFormat: false,
     userInvalidated: false,
     iso: false,
     parsedDateParts: [],
     meridiem: null },
  _locale:
   Locale {
     _ordinalParse: /\d{1,2}(th|st|nd|rd)/,
     ordinal: [Function],
     _abbr: 'en',
     _config:
      { ordinalParse: /\d{1,2}(th|st|nd|rd)/,
        ordinal: [Function],
        abbr: 'en' },
     _ordinalParseLenient: /\d{1,2}(th|st|nd|rd)|\d{1,2}/ },
  _d: 2016-06-11T10:54:24.921Z }

// define an inspect() function
moment.prototype.inspect = function() {
  return this.format('[moment]("YYYY-MM-DD HH:mm:ss.SSSZ")')
}
[Function]

// Easier to read output
> moment()
moment("2016-06-11 11:55:05.192+01:00")

// And it seems to be reverisble
> moment("2016-06-11 11:55:05.192+01:00")
moment("2016-06-11 11:55:05.192+01:00")
```

I'm happy to submit a PR and tests if this is palatable.

**Comment by maggiepint** [comment]
I do all my JS development client side so I don't have a ton of node exposure, but I think this makes sense. Basically what I get from google is that if util.inspect() is called on a moment, node will find this inspect function on the prototype and return that string. Seems like this is largely used for logging purposes?

I will warn that if this affects the ability to read all the properties on the moment if you need to, then it's not a good thing. For debugging purposes, we frequently need to look at some of the internal properties. As long as it doesn't break that I'm cool with it.

**Comment by glenjamin** [comment]
It'll kick in whenever a moment get passed to node's `util.inspect`, which is used internally by `console.log`. All the properties are still there and accessible. In the browser this has no effect unless called explicitly.

If you wanted to log out a moment with all of it's properties in node after these changes, you'd need to do `console.log("%j", m)` or similar.

**Comment by maggiepint** [comment]
Closed in favor of pull #3231
